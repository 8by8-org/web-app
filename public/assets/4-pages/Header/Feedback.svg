<svg width="26" height="22" viewBox="0 0 26 22" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10.7245 5.71704C9.04453 5.71704 8.01367 7.18613 8.01367 8.39122C8.01367 10.9968 10.9752 13.4153 13.0656 14.9549C15.156 13.4152 18.1176 10.9966 18.1176 8.39122C18.1176 7.18623 17.0867 5.71704 15.4068 5.71704C14.4686 5.71704 13.6671 6.46504 13.0657 7.17566C12.4643 6.46504 11.6627 5.71704 10.7246 5.71704H10.7245Z" fill="white"/>
<path d="M13.2489 2.88914e-05C6.21654 2.88914e-05 0.52501 4.4328 0.52501 9.85075C0.52501 11.7388 1.20909 13.5721 2.52247 15.1592C2.00252 16.117 1.31844 17.2662 0.497549 18.1692C-0.0771737 18.826 -0.159214 19.7015 0.27862 20.4677C0.634399 21.097 1.26369 21.4528 1.97508 21.4528C2.08445 21.4528 2.22131 21.4528 2.33086 21.4255C4.90304 20.9603 7.33831 19.9752 9.03482 19.1816C10.403 19.5647 11.8259 19.7288 13.2762 19.7288C20.3085 19.7288 26 15.296 26 9.87808C25.9727 4.43282 20.2537 0 13.2488 0L13.2489 2.88914e-05ZM13.2489 17.704C11.8534 17.704 10.5125 17.5123 9.22648 17.1292C8.98025 17.0472 8.73404 17.0745 8.48763 17.184C6.92783 17.9502 4.54729 18.9353 2.05729 19.4004C3.15172 18.1416 4.02744 16.5547 4.60197 15.46C4.7936 15.077 4.73882 14.6118 4.43782 14.3108C3.17902 12.9974 2.52245 11.4376 2.52245 9.85056C2.52245 5.52733 7.31099 1.99735 13.2488 1.99735C19.1866 1.99735 23.9751 5.52728 23.9751 9.85056C23.9751 14.2014 19.1591 17.7038 13.2488 17.7038L13.2489 17.704Z" fill="white"/>
</svg>
