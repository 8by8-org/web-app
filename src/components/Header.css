.navbar {
  background-color: white;
  height: 60px;
  padding: 10px 30px;
  display: flex;
  flex-direction: row;
}

#icons-tray {
  display: flex;
  flex-direction: row;
  width: 80px;
  justify-content: space-between;
}

.menu-icons {
  font-size: 30px;
}

a.nav-link {
  padding: 0;
  color: black;
}

a.nav-link:hover {
  color: black;
}

/* Sidebar */
.nav-menu {
  background-color: white;
  color: black;
  max-width: 80vw;
  width: 400px;
  height: 100vh;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
  right: -100%;
  transition: 1200ms;
}

.nav-menu.active {
  right: 0;
  transition: 800ms;
}

.menu-header {
  width: 100%;
  height: 60px;
  display: flex;
  justify-content: space-between;
  font-size: 30px;
}

.menu-items {
  width: 100%;
  overflow: auto;
  padding: 20px;
}

.menu-greeting {
  font-size: 30px;
  margin: 30px 0;
  text-transform: uppercase;
}

.nav-text {
  display: flex;
  align-items: center;
  text-align: left;
  list-style: none;
  height: 60px;
  text-transform: uppercase;
  border-radius: 10px;
  padding: 0 5px;
}

.nav-text:hover {
  background-color: rgba(130, 130, 130, 0.25);
}

.nav-text a {
  text-decoration: none;
  font-size: 18px;
  width: 95%;
  height: 100%;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-radius: 4px;
  padding: 0;
}

.nav-text p {
  margin: 0;
  margin-left: 10px;
}

.menu-settings {
  margin: 10px 0;
}

.menu-settings a.nav-link {
  margin: 10px 0;
}

/* Notifications tray */
.notif-menu {
  height: 80vh;
  width: 100vw;
  max-width: 500px;
  position: fixed;
  bottom: -100%;
  transition: 1200ms;
  background-color: white;
  color: black;
  border-top: 2px solid black;
  width: 100vw;
}

.notif-menu.active {
  bottom: 0;
  transition: 800ms;
}

.notif-toggle {
  width: 100%;
  height: 60px;
  display: flex;
}

.notif-menu-items {
  width: 100%;
  overflow: auto;
  padding: 10px;
}

.notif-title {
  margin: auto;
  font-size: 30px;
}

.notif-container {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  width: 100%;
  padding: 10px 20px;
}

.false {
  font-weight: bolder;
}

.notif-icon {
  display: flex;
  align-items: center;
  width: 12.5%;
  text-align: center;
  font-size: 30px;
}

.notif-text {
  width: 65%;
}

.notif-date {
  width: 22.5%;
  text-align: end;
}
