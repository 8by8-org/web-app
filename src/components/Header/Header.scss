.navbar {
	background-color: black;
	height: 57px;
	padding: 0px 30px 0px 40px;
	overflow: hidden;
	position: fixed; /* Set the navbar to fixed position */
	top: 0; /* Position the navbar at the top of the page */
	width: 100%; /* Full width */
}

#brand-logo {
	height: 29px;
	padding: 0;
}

#icons-tray {
	height: 35px;
	width: 80px;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
}

#sidebar-icon {
	font-size: 34px;
	position: relative;
	bottom: 5px;
}

#notif-icon {
	font-size: 30px;
	position: relative;
	bottom: 4px;
}

a.nav-link {
	padding: 0;
	color: white;
}

a.nav-link:hover {
	color: white;
}

a.navbar-brand {
	padding: 0;
}

/* Sidebar */
.nav-menu {
	background-color: black;
	color: white;
	max-width: 80vw;
	width: 380px;
	height: 100vh;
	display: flex;
	justify-content: center;
	position: fixed;
	top: 0;
	right: -380px;
	transition: 400ms;
}

.nav-menu.active {
	right: 0;
	transition: 400ms;
}

.menu-header {
	width: 100%;
	height: 60px;
	position: static;
	display: flex;
	justify-content: space-between;
	font-size: 30px;
}

.menu-items {
	width: 100%;
	overflow: auto;
	padding: 20px 30px;
}

.menu-greeting {
	font-size: 30px;
	margin: 30px 0;
	text-transform: uppercase;
}

.nav-text {
	display: flex;
	align-items: center;
	text-align: left;
	list-style: none;
	height: 60px;
	text-transform: uppercase;
	border-radius: 10px;
	padding: 0 5px;
}

.nav-text:hover {
	background-color: rgba(130, 130, 130, 0.25);
}

.nav-text a {
	text-decoration: none;
	font-size: 18px;
	width: 95%;
	height: 100%;
	display: flex;
	align-items: center;
	flex-direction: row;
	border-radius: 4px;
	padding: 0;
}

.nav-text p {
	margin: 0;
	margin-left: 10px;
}

.menu-settings {
	margin: 10px 0;
}

.menu-settings a.nav-link {
	margin: 10px 0;
}

/* Notifications tray */
.notif-wrapper {
	width: 100vw;
	display: flex;
	/* align-items: center; */
	justify-content: center;
}

.notif-menu {
	display: flex;
	height: 80vh;
	width: 100vw;
	max-width: 500px;
	position: fixed;
	bottom: -80vh;
	transition: 400ms;
	background-color: black;
	color: white;
	border-top: 2px solid black;
	width: 100vw;
}

.notif-menu.active {
	bottom: 0;
	transition: 400ms;
}

.notif-toggle {
	width: 100%;
	height: 60px;
	display: flex;
}

#close-icon {
	font-size: 30px;
	position: absolute;
	top: 14px;
	left: 15px;
}

.notif-items {
	width: 100%;
	overflow: auto;
	padding: 10px;
}

.notif-title {
	margin: auto;
	font-size: 30px;
}

.notif-container {
	display: flex;
	flex-direction: row;
	justify-content: flex-start;
	width: 100%;
	padding: 10px 20px;
}

.false {
	font-weight: bolder;
}

.notif-icon {
	display: flex;
	align-items: center;
	width: 12.5%;
	text-align: center;
	font-size: 30px;
}

.notif-text {
	width: 65%;
}

.notif-date {
	width: 22.5%;
	text-align: end;
}
