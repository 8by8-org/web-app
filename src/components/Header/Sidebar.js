import React, { useEffect, useState } from "react";
import "./Sidebar.scss";
import { useAuth } from "../../contexts/AuthContext";
import { IoIosArrowForward } from "react-icons/io";
import { getUserDatabase } from "./../../functions/UserData.js";
import Blob from "./../../assets/4-pages/Sidebar/Blob.svg";
import PopupModal from "../PopupModal/PopupModal";

function Sidebar({ sidebar, showSidebar }) {
  const { currentUser } = useAuth();
  const [userData, setUserData] = useState();
  const [openSignoutModal, setSignoutModal] = useState(false);
  const [userType, setUserType] = useState("none");

  useEffect(() => {
    if (currentUser) {
      getUserDatabase().then((data) => {
        setUserData(data);

        if (data.startedChallenge && data.invitedBy.length > 0) {
          setUserType("hybrid");
        } else if (data.startedChallenge) {
          setUserType("challenger");
        } else {
          setUserType("player");
        }
      });
    }
  }, [sidebar]);

  function Greeting() {
    return (
      <>
        {userData ? (
          <div className="avatar-greeting">
            <div className="blob">
              <img src={Blob} />
            </div>
            {userData.avatar ? (
              <img
                className="avatar"
                src={
                  require(`./../../assets/3-avatars/avatar-${userData.avatar}.svg`)
                    .default
                }
              />
            ) : (
              <img
                className="avatar"
                src={require(`./../../assets/3-avatars/avatar-1.svg`).default}
              />
            )}
            <div className="text">Hi {userData.name}!</div>
          </div>
        ) : (
          <div className="text">Hi There!</div>
        )}
      </>
    );
  }

  function Links() {
    return (
      <div className="container">
        {userType === "none" && (
          <>
            <a onClick={showSidebar} href="/challengerwelcome">
              <button className="small gradient">
                <span>Take The Challenge</span>
              </button>
            </a>
            <a onClick={showSidebar} href="/actions">
              Take Action
            </a>
          </>
        )}
        {userType === "player" && (
          <>
            <a onClick={showSidebar} href="/challengerwelcome">
              <button className="small gradient">
                <span>Take The Challenge</span>
              </button>
            </a>
            <a onClick={showSidebar} href="/actions">
              Take Action
            </a>
          </>
        )}
        {userType === "challenger" && (
          <>
            <a onClick={showSidebar} href="/progress">
              My Challenge
            </a>
          </>
        )}
        {userType === "hybrid" && (
          <>
            <a onClick={showSidebar} href="/progress">
              My Challenge
            </a>
            <a onClick={showSidebar} href="/actions">
              Take Action
            </a>
          </>
        )}
      </div>
    );
  }

  return (
    <div className="sidebar">
      <div className="top" onClick={showSidebar}>
        <IoIosArrowForward />
      </div>

      <div className="greeting">
        <Greeting />
      </div>

      <div className="links">
        <Links />
      </div>

      <div className="bottom-links">
        <a onClick={showSidebar} href="/privacypolicy">
          Privacy Policy
        </a>
        {currentUser ? (
          <a onClick={() => setSignoutModal(true)}>Sign Out</a>
        ) : (
          <a onClick={showSidebar} href="/signup">
            Sign Up
          </a>
        )}
      </div>

      {openSignoutModal && (
        <PopupModal
          setOpenModal={setSignoutModal}
          content={
            <>
              <div className="b2">Are you sure you want to sign out?</div>

              <div className="buttons">
                <a href="/signout">
                  <button
                    className="gradient small"
                    onClick={() => {
                      setSignoutModal(false);
                    }}
                  >
                    <span>Yes, but I'll be back</span>
                  </button>
                </a>
              </div>

              <div className="buttons">
                <button
                  className="text-gradient small"
                  onClick={() => {
                    setSignoutModal(false);
                  }}
                >
                  <span>No, I think I'll stay</span>
                </button>
              </div>
            </>
          }
        />
      )}
    </div>
  );
}

export default Sidebar;
