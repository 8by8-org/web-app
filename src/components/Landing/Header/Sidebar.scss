@import "src/styles/main";

.sidebar {
  .top {
    svg {
      width: 35px;
      height: 35px;
      position: relative;
      right: 10px;
    }
    :hover {
      cursor: pointer;
    }
  }

  .greeting {
    margin: 50px 0;
    .text {
      font-family: $bebas-neue;
      font-weight: normal;
      font-size: 36px;
    }
    .avatar-greeting {
      display: flex;
      align-items: center;
      .blob {
        width: 0;
        img {
          width: 80px;
        }
      }
      .avatar {
        width: 65px;
        position: relative;
        left: 8px;
      }
      .text {
        margin-left: 30px;
      }
    }
  }

  .links,
  .bottom-links {
    margin-left: 20px;

    button {
      position: relative;
      right: 24px;
      font-size: 28px;
    }
  }

  .links {
    .container {
      padding: 0;
      margin-bottom: 48px;
      display: flex;
      flex-direction: column;
      a {
        color: white;
        text-decoration: none;
        margin-bottom: 24px;
        font-family: $bebas-neue;
        font-weight: normal;
        font-size: 28px;
      }
    }
  }

  .bottom-links {
    display: flex;
    flex-direction: column;
    button {
      background-color: transparent;
      border: none;
      cursor: pointer;
      text-decoration: none;
      display: inline;
      margin: 5px 0px;
      padding: 0;
      color: white;
      font-weight: 700;
      font-size: 18px;
      line-height: 21.6px;
      text-align: left;
      right: 0px;
    }
    
    a {
      padding: 0;
      color: white;
      text-decoration: none;
      margin: 5px 0;
      font-weight: 700;
      font-size: 18px;
      line-height: 21.6px;
    }
  }

  .popup-modal {
    .buttons {
      width: 100%;
      a {
        text-decoration: none;
      }
      button {
        margin: auto;
        margin-top: 35px;
        margin-bottom: 35px;
      }
    }
  }
}
