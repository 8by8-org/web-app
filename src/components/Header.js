import React, { useState, useEffect } from "react";
import { Nav, Navbar } from "react-bootstrap";
import { useAuth } from "../contexts/AuthContext";
import { IconContext } from "react-icons";
import * as FaIcons from "react-icons/fa";
import * as GiIcons from "react-icons/gi";
import * as IoIcons from "react-icons/io";
import * as MdIcons from "react-icons/md";
import logo from "../assets/logos/logo.svg";
import sidebarLogo from "../assets/logos/black-logo.svg";
import "./Header.scss";
import { db } from "../firebase";

function Header() {
  const [sidebar, setSidebar] = useState(false);
  const showSidebar = () => {
    setSidebar(!sidebar);
    setNotif(false);
  };

  const [notif, setNotif] = useState(false);
  const showNotif = () => {
    setNotif(!notif);
    setSidebar(false);
  };

  const { currentUser } = useAuth();
  const greeting = `Hi ${currentUser.name ? currentUser.name : "There"}!`;

  // all sidebar links lead to path: /

  const SidebarData = {
    "Take the Challenge": {
      title: "Take the Challenge",
      path: "/",
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
    "My Challenge": {
      title: "My Challenge",
      path: `/progress/${currentUser?.uid}`,
      icon: <GiIcons.GiJeweledChalice />,
      show: false
    },
    "Take Action": {
      title: "Take Action",
      path: `/actions/${currentUser?.uid}`,
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
    "Notifications": {
      title: "Notifications",
      path: "/",
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
    "Share": {
      title: "Share",
      path: "/",
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
    "Why 8by8?": {
      title: "Why 8by8?",
      path: "/",
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
    "FAQS": {
      title: "FAQS",
      path: "/",
      icon: <GiIcons.GiJeweledChalice />,
      show: true
    },
  };

  const NotifData = [
    {
      icon: <IoIcons.IoIosPerson />,
      text: "Olivia sent you a 🎊 for setting election reminders",
      date: "Just now",
      read: "false",
    },
    {
      icon: <IoIcons.IoIosPerson />,
      text: "Elliot took your challenge invite",
      date: "Just now",
      read: "false",
    },
    {
      icon: <IoIcons.IoIosPerson />,
      text: "Fin took your challenge invite",
      date: "Aug 9",
      read: "true",
    },
    {
      icon: <IoIcons.IoIosPerson />,
      text: "Fin sent you a 👏 for registering to vote",
      date: "Aug 9",
      read: "true",
    },
  ];

  const [sideBarDynamic, setSideBarDynamic] = useState(SidebarData);
  
  // sets challange started status for sideBar buttons
  useEffect(() => {
    if (currentUser?.uid){
      db.collection("users").doc(currentUser?.uid).collection("challenge").doc("challenge")
        .get()
        .then(doc => {
          if (doc.exists) {
            setSideBarDynamic({
              ...sideBarDynamic,
              "Take the Challenge": {
                ...sideBarDynamic["Take the Challenge"],
                "show": false
              },
              "My Challenge": {
                ...sideBarDynamic["My Challenge"],
                "challengeStarted": true,
              }
            })
          }
          else {
            setSideBarDynamic({
              ...sideBarDynamic,
              "Take the Challenge": {
                ...sideBarDynamic["Take the Challenge"],
                "show": false
              },
              "My Challenge": {
                ...sideBarDynamic["My Challenge"],
                "challengeStarted": false,
              }
            })
          }
        });
    }
  }, [currentUser?.uid])
  
  return (
    <>
      <IconContext.Provider value={{ color: "black" }}>
        <div className="navbar">
          <Navbar.Brand href="/">
            <img src={logo} alt="8by8 logo" id="brand-logo" />
          </Navbar.Brand>
          <div id="icons-tray">
            <Nav.Link to="#" id="sidebar-icon">
              <MdIcons.MdNotificationsNone onClick={showNotif} />
            </Nav.Link>
            <Nav.Link to="#" id="notif-icon">
              <FaIcons.FaBars onClick={showSidebar} />
            </Nav.Link>
          </div>
        </div>

        {/* Sidebar */}
        <nav className={sidebar ? "nav-menu active" : "nav-menu"}>
          <ul className="menu-items" onClick={showSidebar}>
            <li className="menu-header">
              <Nav.Link to="#">
                <IoIcons.IoIosArrowForward />
              </Nav.Link>
              <img src={sidebarLogo} alt="8by8 logo" />
            </li>
            <p className="menu-greeting">{greeting}</p>
            {Object.values(sideBarDynamic).map((item, index) => {
              if(item.show){
                return (
                  <li key={index} className="nav-text">
                    <Nav.Link href={item.path}>
                      {item.icon}
                      <p>{item.title}</p>
                    </Nav.Link>
                  </li>
                )
              };
            })}
            <div className="menu-settings">
              <Nav.Link href="/">Settings</Nav.Link>
              <Nav.Link href="/">Privacy Policy</Nav.Link>
              {currentUser ? (
                <Nav.Link href="/logout">Logout</Nav.Link>
              ) : (
                <Nav.Link href="/login">Login</Nav.Link>
              )}
            </div>
          </ul>
        </nav>

        {/* Notifications */}
        <div className="notif-wrapper">
          <nav className={notif ? "notif-menu active" : "notif-menu"}>
            <ul className="notif-items" onClick={showNotif}>
              <li className="notif-toggle">
                <Nav.Link to="#" id="close-icon">
                  <MdIcons.MdClose />
                </Nav.Link>
                <span className="notif-title">Notifications</span>
              </li>
              {NotifData.map((item, index) => {
                return (
                  <li key={index} className={"notif-container " + item.read}>
                    <span className="notif-icon">{item.icon}</span>
                    <span className="notif-text">{item.text}</span>
                    <span className="notif-date">{item.date}</span>
                  </li>
                );
              })}
            </ul>
          </nav>
        </div>
      </IconContext.Provider>
    </>
  );
}

export default Header;
