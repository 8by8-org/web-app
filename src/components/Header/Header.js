import React, { useState } from "react";
import { Nav, Navbar } from "react-bootstrap";
import { IconContext } from "react-icons";
import * as FaIcons from "react-icons/fa";
import feedback from "./../../assets/4-pages/Header/Feedback.svg";
import logo from "./../../assets/logos/white-logo.svg";
import "./Header.scss";
import Sidebar from "./Sidebar";

function Header({ isShare }) {
  // Open and closes sidebar and closes the notification pop-up.
  const [sidebar, setSidebar] = useState(false);
  const showSidebar = () => {
    setSidebar(!sidebar);
  };

  return (
    <>
      <IconContext.Provider value={{ color: "white" }}>
        <div className="navbar" id={isShare ? "navbar-share" : "navbar"}>
          <div className="navbar-buttons">
            <Navbar.Brand href={isShare ? "#" : "/"}>
              <img src={logo} alt="8by8 logo" id="brand-logo" />
            </Navbar.Brand>
            <div id="icons-tray">
              <button
                onClick={() =>
                  !isShare &&
                  window.open("https://forms.gle/r33L2NAKT69MrvsZ7", "_blank")
                }
                id="feedback-icon"
              >
                <img src={feedback} />
              </button>

              <Nav.Link to="#" id="sidebar-icon">
                <FaIcons.FaBars onClick={showSidebar} />
              </Nav.Link>
            </div>
          </div>
        </div>

        {/* Sidebar (don't render when isShare is true, when used in preview) */}
        {!isShare && (
          <div className={sidebar ? "nav-container active" : "nav-container"}>
            <nav className="nav-menu">
              <ul className="menu-items">
                <Sidebar sidebar={sidebar} showSidebar={showSidebar} />
              </ul>
            </nav>
          </div>
        )}
      </IconContext.Provider>
    </>
  );
}

export default Header;
