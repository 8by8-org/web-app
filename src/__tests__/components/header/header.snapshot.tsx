import { render, cleanup } from '@testing-library/react'
import Header from '@/components/header/header';

describe('Header', () => {
  //add showModal and close to HTMLDialogElement.prototype as they are not recognized
  //as properties of this type of element by the testing framework
  HTMLDialogElement.prototype.showModal = () => {};
  HTMLDialogElement.prototype.close = () => {};
  afterEach(cleanup);

  it('renders the header unchanged', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
