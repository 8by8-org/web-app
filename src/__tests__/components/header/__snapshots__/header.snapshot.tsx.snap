// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header unchanged 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header_content"
    >
      <a
        class="brand_link"
        href="/"
      >
        <img
          alt="8by8 logo"
          class="brand_logo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
      <div
        class="icons_tray"
      >
        <button
          aria-label="open feedback form"
          class="feedback_btn"
        >
          <img
            alt="feedback icon"
            class="feedback_btn_icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
        <button
          aria-label="open navigation menu"
          class="hamburger_btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: white;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
  <div
    class="hidden"
  >
    <div
      class="hamburger_menu_inner_container"
    >
      <nav
        class="hamburger_menu_closing"
      >
        <ul
          class="hamburger_menu_items"
        >
          <li>
            <button
              aria-label="close navigation menu"
              class="close_hamburger_btn"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
                />
              </svg>
            </button>
          </li>
          <h2
            class="standalone_greeting"
          >
            Hi there!
          </h2>
          <li>
            <a
              class="take_the_challenge_btn"
              href="/challengerwelcome"
            >
              Take The Challenge
            </a>
          </li>
          <li>
            <a
              class="link_lg_top"
              href="/actions"
            >
              Take Action
            </a>
          </li>
          <li>
            <a
              class="link_lg"
              href="/why8by8"
            >
              Why 8by8
            </a>
          </li>
          <li>
            <a
              class="link_lg"
              href="/rewards"
            >
              Rewards
            </a>
          </li>
          <li>
            <a
              class="link_lg"
              href="/faq"
            >
              FAQS
            </a>
          </li>
          <li>
            <a
              class="link_sm_top"
              href="/privacy"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              class="link_sm"
              href="/settings"
            >
              Settings
            </a>
          </li>
          <li>
            <a
              class="link_sm"
              href="/signup"
            >
              Sign up
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <dialog
    aria-label="Are you sure you want to sign out?"
    class="popup_modal_dark_theme"
    role="dialog"
  >
    <div>
      <div
        class="close_btn_container"
      >
        <button
          aria-label="close dialog"
          class="close_btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: white;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
      </div>
      <div
        class="content"
      >
        <p
          class="b1"
        >
          Are you sure you want to sign out?
        </p>
        <button
          class="btn_top"
        >
          <span>
            Yes, but I'll be back
          </span>
        </button>
        <button
          class="btn_bottom"
        >
          <span>
            No, I think I'll stay
          </span>
        </button>
      </div>
    </div>
  </dialog>
</div>
`;
